lib/ansible/modules/cloud/packet/packet_device.py

```diff
@@ -415,35 +437,52 @@ def create_single_device(module, packet_conn, hostname):
     return device
 
 
-def wait_for_ips(module, packet_conn, created_devices):
+def refresh_device_list(module, packet_conn, devices):
+    device_ids = [d.id for d in devices]
+    new_device_list = get_existing_devices(module, packet_conn)
+    return [d for d in new_device_list if d.id in device_ids]
 
-    def has_public_ip(addr_list):
-        return any([a['public'] and (len(a['address']) > 0) for a in addr_list])
 
-    def all_have_public_ip(ds):
-        return all([has_public_ip(d.ip_addresses) for d in ds])
+def wait_for_devices_active(module, packet_conn, watched_devices):
+    wait_timeout = module.params.get('wait_timeout')
+    wait_timeout = time.time() + wait_timeout
+    while wait_timeout > time.time():
+        refreshed = refresh_device_list(module, packet_conn, watched_devices)
+        if all(d.state == 'active' for d in refreshed):
+            return refreshed
+        time.sleep(5)
+    raise Exception("Waiting for state \"active\" timed out for devices: %s"
```

Comment:
When `wait_timeout` is `0` the exception message is `Waiting for state \"active\" timed out for devices: []"`, what do you think of initializing `refreshed` with `watched_devices` instead of an empty list ?